### PR TITLE
[OPS-824] Add automatic deployment via S3/cloudfront

### PR DESCRIPTION
**Pull request by yorickvP**
currently already up at https://ghc.dev/, but the deployment step has been poorly tested

From @int-index, I'd like some more information about the eventual files we're deploying, maybe make a separate output path.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+steps:
+  - commands:
+    - "nix-shell --run \"cabal v2-update && make\""
+    label: "make"
+    artifact_paths: "*.html"
+  - wait
+  - label: 'Deploy'
+    commands:
+    - "buildkite-agent artifact download '*.html'"
```
Comment (mkaito) [inline .buildkite/pipeline.yml]: :frowning_face: god forbid there is even a CSS file or something.

File: deployment/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+This dir contains terraform information for the AWS resources that are needed for the ghc.dev hosting.
+The state is stored in an aws bucket on our account.
+
```
Comment (mkaito) [inline deployment/README.md]: Mention the need for an API TOKEN and to set `AWS_PROFILE` environment variable.

File: deployment/shell.nix [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{ pkgs ? import <nixpkgs> {} }:
+pkgs.mkShell {
+  buildInputs = [ pkgs.terraform-full ];
```
Comment (mkaito) [inline deployment/shell.nix]: I mean, that works, but maybe more explicit to use `terraform.withPlugins`?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (yorickvP) [reply .buildkite/pipeline.yml]: Yeah, I would like it if there was an output dir

**Pull request merged**